core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -187,26 +188,60 @@ private static boolean isComplete(int flags) {
 
     @Override
     public boolean isRequestComplete() {
-        return isAvailable(flags, RequestLogProperty.FLAGS_REQUEST_COMPLETE);
+        return hasInterestedFlags(flags, RequestLogProperty.FLAGS_REQUEST_COMPLETE);
     }
 
     @Override
     public boolean isAvailable(RequestLogProperty property) {
-        return isAvailable(flags, property.flag());
+        requireNonNull(property, "property");
+        return hasInterestedFlags(flags, property.flag());
+    }
+
+    @Override
+    public boolean isAvailable(RequestLogProperty... properties) {
+        requireNonNull(properties, "properties");
+        checkArgument(properties.length != 0, "properties is empty.");
+        return isAvailable(interestedFlags(properties));
+    }
+
+    @Override
+    public boolean isAvailable(Iterable<RequestLogProperty> properties) {
+        requireNonNull(properties, "properties");
+        final int flags = interestedFlags(properties);
+        checkArgument(flags != 0, "properties is empty.");
+        return isAvailable(flags);
     }
 
     private boolean isAvailable(int interestedFlags) {
```

Comment:
nit: can remove this method and just call `hasInterestedFlags` directly?

Reply:
I guess it's better keeping it for readability - `isAvailable(interestedFlags)` vs. `hasInterestedFlags(this.flags, interestedFlags)`.

Reply:
SGTM. 😄